src/mam4xx/aer_rad_props.hpp

```diff
@@ -33,6 +33,98 @@ constexpr Real cnst_faktor =
+KOKKOS_INLINE_FUNCTION
+void volcanic_cmip_sw2(const ConstColumnView &zi, const int ilev_tropp,
+                       const View2D &ext_cmip6_sw_inv_m,
+                       const View2D &ssa_cmip6_sw, const View2D &af_cmip6_sw,
+                       const View2D &tau, const View2D &tau_w,
+                       const View2D &tau_w_g, const View2D &tau_w_f) {
+
+  // Intent-in
+  //  ncol       ! Number of columns
+  //  zi(:,:)    ! Height above surface at interfaces [m]
+  //  trop_level(pcols)  ! tropopause level index
+  //  ext_cmip6_sw_inv_m(pcols,pver,nswbands)  ! short wave extinction [m^{-1}]
+  //  ssa_cmip6_sw(:,:,:),af_cmip6_sw(:,:,:)
+
+  // Intent-inout
+  // Note: we are using emaxx layouts
+  //  tau    (pcols,nswbands,0:pver) ! aerosol extinction optical depth
+  //  tau_w  (pcols,nswbands,0:pver) ! aerosol single scattering albedo * tau
+  //  tau_w_g(pcols,nswbands,0:pver) ! aerosol assymetry parameter * tau * w
+  //  tau_w_f(pcols,nswbands,0:pver) ! aerosol forward scattered fraction * tau
+  //  *
+  //  w
+
+  // !Local variables
+  // icol, ipver, ilev_tropp
+  // lyr_thk ! thickness between level interfaces [m]
+  // ext_unitless(nswbands), asym_unitless(nswbands)
+  // ext_ssa(nswbands),ext_ssa_asym(nswbands)
+
+  // !Logic:
+  // !Update taus, tau_w, tau_w_g and tau_w_f with the read in volcanic
+  // !aerosol extinction (1/km), single scattering albedo and asymmtry factors.
+
+  // !Above the tropopause, the read in values from the file include both the
+  // stratospheric !and volcanic aerosols. Therefore, we need to zero out taus
+  // above the tropopause !and populate them exclusively from the read in
+  // values.
+
+  // !If tropopause is found, update taus with 50% contributuions from the
+  // volcanic input !file and 50% from the existing model computed values
+
+  // !First handle the case of tropopause layer itself:
+  // do icol = 1, ncol
+  // ilev_tropp = trop_level(icol) !tropopause level
+  //
+  constexpr Real half = 0.5;
+  const Real lyr_thk = zi(ilev_tropp) - zi(ilev_tropp + 1);
+  for (int i = 0; i < nswbands; ++i) {
+    // NOTE: shape of ext_cmip6_sw_inv_m (nswbands,pver)
+    const Real ext_unitless = lyr_thk * ext_cmip6_sw_inv_m(i, ilev_tropp);
+    const Real asym_unitless = af_cmip6_sw(ilev_tropp, i);
+    const Real ext_ssa = ext_unitless * ssa_cmip6_sw(ilev_tropp, i);
+    const Real ext_ssa_asym = ext_ssa * asym_unitless;
+    // NOTE: tau vars have one extra dimension in Fortran.
+    tau(i, ilev_tropp + 1) = half * (tau(i, ilev_tropp + 1) + ext_unitless);
+    tau_w(i, ilev_tropp + 1) = half * (tau_w(i, ilev_tropp + 1) + ext_ssa);
+    tau_w_g(i, ilev_tropp + 1) =
+        half * (tau_w_g(i, ilev_tropp + 1) + ext_ssa_asym);
+    tau_w_f(i, ilev_tropp + 1) =
+        half * (tau_w_f(i, ilev_tropp + 1) + ext_ssa_asym * asym_unitless);
+  } // end i
+
+  // !As it will be more efficient for FORTRAN to loop over levels and then
```

Comment:
Which previous comment does not apply? Do we need line breaks before each Fortran comment delimiter (`!`)?

Reply:
I will delete this comment.